discovery_engine_core/core/src/document.rs

```diff
@@ -107,3 +107,33 @@ where
 
 /// A 1-dimensional sequence embedding.
 pub type Embedding1 = Embedding<Ix1>;
+
+/// Log the time that has been spent on the document.
+pub struct TimeLogged {
```

Comment:
It is the name we already have in dart. But I can change it in both places. @finiteprods

Reply:
I think `TimeLogged` is ok actually because this struct is supposed to represent the _event_ "a view-time is logged". maybe for extra clarity it could be called `ViewtimeLogged`, or `TimeIntervalLogged`. up to you.

Reply:
I went for `TimeSpent`